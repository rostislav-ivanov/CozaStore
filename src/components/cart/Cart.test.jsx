import { test, describe, expect, beforeEach } from "vitest";
import { render, screen, prettyDOM, fireEvent } from "../../tests/test-utils";

import Cart from "./Cart";
import {
  mockBagContextValue,
  mockBagDate,
  mockCount,
} from "../../tests/mockData";

describe("Cart component", () => {
  beforeEach(() => {
    mockBagContextValue.bag = [...mockBagDate];
    mockBagContextValue.count = mockCount;
  });

  test("renders correctly", () => {
    const { container } = render(<Cart />);
    //console.log(prettyDOM(container));
    expect(container).toMatchSnapshot();
  });

  test("displays items in the cart", () => {
    render(<Cart />);
    mockBagDate.forEach((item) => {
      expect(screen.getByText(`Color: ${item.color}`)).toBeInTheDocument();
      expect(screen.getByText(`Size: ${item.size}`)).toBeInTheDocument();
      expect(screen.getByText(item.name)).toBeInTheDocument();
      expect(
        screen.getByText(
          `${item.quantity} x $ ${Number(item.price).toFixed(2)}`
        )
      ).toBeInTheDocument();
    });
  });

  test("displays correct total amount", () => {
    const total = mockBagDate.reduce(
      (acc, item) => acc + item.price * item.quantity,
      0
    );
    render(<Cart />);
    expect(
      screen.getByText(`Total: $ ${Number(total).toFixed(2)}`)
    ).toBeInTheDocument();
  });

  test("handles empty cart", () => {
    mockBagContextValue.bag = [];
    mockBagContextValue.count = 0;
    render(<Cart />);
    expect(screen.getByText("Your cart is empty!")).toBeInTheDocument();
  });

  test("checkout button is present and clickable", () => {
    render(<Cart />);
    const checkoutButton = screen.getByText("Check Out");
    expect(checkoutButton).toBeInTheDocument();
    fireEvent.click(checkoutButton);

    expect(checkoutButton.closest("a")).toHaveAttribute("href", "/checkout");
  });
});
