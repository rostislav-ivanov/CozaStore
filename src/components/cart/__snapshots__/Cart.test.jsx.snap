// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Cart component > renders correctly 1`] = `
<div>
  <div
    class="wrap-header-cart js-panel-cart show-header-cart"
  >
    <div
      class="s-full js-hide-cart"
    />
    <div
      class="header-cart flex-col-l p-l-65 p-r-25"
    >
      <div
        class="header-cart-title flex-w flex-sb-m p-b-8"
      >
        <span
          class="mtext-103 cl2"
        >
          Your Cart
        </span>
        <div
          class="fs-35 lh-10 cl2 p-lr-5 pointer hov-cl1 trans-04"
        >
          <i
            class="zmdi zmdi-close"
          />
        </div>
      </div>
      <div
        class="header-cart-content flex-w js-pscroll"
      >
        <ul
          class="header-cart-wrapitem w-full"
        >
          <li
            class="header-cart-item flex-w flex-t m-b-12"
          >
            <div
              class="header-cart-item-img"
            >
              <img
                alt="IMG"
                src="/images/product-01.jpg"
              />
            </div>
            <div
              class="header-cart-item-txt"
            >
              <a
                class="header-cart-item-name hov-cl1 trans-04"
                href="/details/53d4dbf5-7f41-47ba-b485-43eccb91cb95"
              >
                Esprit Ruffle Shirt
              </a>
              <span
                class="header-cart-item-info"
              >
                Color: 
                Red
              </span>
              <span
                class="header-cart-item-info"
              >
                Size: 
                M
              </span>
              <span
                class="header-cart-item-info"
              >
                2
                 x $ 
                16.64
              </span>
            </div>
          </li>
          <li
            class="header-cart-item flex-w flex-t m-b-12"
          >
            <div
              class="header-cart-item-img"
            >
              <img
                alt="IMG"
                src="/images/product-03.jpg"
              />
            </div>
            <div
              class="header-cart-item-txt"
            >
              <a
                class="header-cart-item-name hov-cl1 trans-04"
                href="/details/c7f51805-242b-45ed-ae3e-80b68605141b"
              >
                Only Check Trouser
              </a>
              <span
                class="header-cart-item-info"
              >
                Color: 
                Blue
              </span>
              <span
                class="header-cart-item-info"
              >
                Size: 
                S
              </span>
              <span
                class="header-cart-item-info"
              >
                3
                 x $ 
                25.50
              </span>
            </div>
          </li>
        </ul>
        <div
          class="w-full"
        >
          <div
            class="header-cart-total w-full p-tb-40"
          >
            Total: $ 
            109.78
          </div>
          <div
            class="header-cart-buttons flex-w w-full"
          >
            <a
              class="flex-c-m stext-101 cl0 size-107 bg3 bor2 hov-btn3 p-lr-15 trans-04 m-r-8 m-b-10"
              href="/checkout"
            >
              Check Out
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
